File: src/main/java/org/zenbot/szolnok/timetable/batch/step/bus/processor/StartBusStopSelectorItemProcessorHelper.java
package org.zenbot.szolnok.timetable.batch.step.bus.processor;

import org.jsoup.nodes.Document;
import org.springframework.stereotype.Component;
import org.zenbot.szolnok.timetable.configuration.properties.TimetableSelectorProperties;

@Component
public class StartBusStopSelectorItemProcessorHelper {

    private final StringCleaner stringCleaner;

    public StartBusStopSelectorItemProcessorHelper(StringCleaner stringCleaner) {
        this.stringCleaner = stringCleaner;
    }

    public String getStartBusStop(Document htmlDocument, TimetableSelectorProperties selectorProperties) {
        return stringCleaner.clean(htmlDocument.select(selectorProperties.getFromSelector()).text());
    }
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/batch/step/bus/processor/StartBusStopSelectorItemProcessorHelper.kt
package org.zenbot.szolnok.timetable.batch.bus.batch.step.bus.processor

import org.jsoup.nodes.Document
import org.springframework.stereotype.Component
import org.zenbot.szolnok.timetable.batch.utils.common.properties.TimetableSelectorProperties
import org.zenbot.szolnok.timetable.batch.utils.common.util.StringCleaner

@Component
class StartBusStopSelectorItemProcessorHelper(private val stringCleaner: StringCleaner) {

    fun getStartBusStop(htmlDocument: Document, selectorProperties: TimetableSelectorProperties): String {
        return stringCleaner.clean(htmlDocument.select(selectorProperties.fromSelector).text())
    }
}
